src/root.jl

```diff
@@ -364,9 +364,13 @@ function auto_T_JaggT(f::ROOTFile, branch; customstructs::Dict{String, Type})
                 elseif elname == "unsigned int" 
                     UInt32
                 elseif elname == "unsigned char" 
-                    Char
+                    UInt8
```

Comment:
what's wrong with `Char`?

Reply:
without that change, I was getting
```julia
julia> ROOTFile("tree_basictypes.root")["t/unsignedchar"]
3-element LazyBranch{Vector{Char}, UnROOT.Offsetjagg, ArraysOfArrays.VectorOfVectors{Char, Vector{Char}, Vector{Int32}, Vector{Tuple{}}}}:
Error showing value of type LazyBranch{Vector{Char}, UnROOT.Offsetjagg, ArraysOfArrays.VectorOfVectors{Char, Vector{Char}, Vector{Int32}, Vector{Tuple{}}}}:
ERROR: ArgumentError: cannot reinterpret `UInt8` as `Vector{Char}`, type `Vector{Char}` is not a bits type
```
and we have
https://github.com/tamasgal/UnROOT.jl/blob/635ed0e46cc72054357a72d52d11aa4eb77a1f2d/src/root.jl#L289-L291

Reply:
after:
```julia
julia> ROOTFile("tree_basictypes.root")["t/char"]
3-element LazyBranch{Vector{Int8}, UnROOT.Offsetjagg, ArraysOfArrays.VectorOfVectors{Int8, Vector{Int8}, Vector{Int32}, Vector{Tuple{}}}}:
 0-element view(::Vector{Int8}, 1:0) with eltype Int8
 Int8[1]
 Int8[1, 1]

julia> ROOTFile("tree_basictypes.root")["t/unsignedchar"]
3-element LazyBranch{Vector{UInt8}, UnROOT.Offsetjagg, ArraysOfArrays.VectorOfVectors{UInt8, Vector{UInt8}, Vector{Int32}, Vector{Tuple{}}}}:
 0-element view(::Vector{UInt8}, 1:0) with eltype UInt8
 UInt8[0x01]
 UInt8[0x01, 0x01]
```

Reply:
that's actually a bug on the reinterpret, reinterpret should becalled as `reinterpret(eltype, Vector{UInt8})`

Reply:
```julia
Char is a 32-bit AbstractChar type that is the default representation of characters in Julia.
```

nvm, I think we still need `UInt8`

Reply:
Ah right `Char` is fundamentally different (4 bytes vs 1 byte)...
```cpp
root [2] sizeof(char)
(unsigned long) 1
root [3] sizeof(unsigned char)
(unsigned long) 1
root [4] sizeof(int)
(unsigned long) 4
```